src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportProcessingJob.cs

```diff
@@ -74,11 +78,29 @@ public Task<string> ExecuteAsync(JobInfo jobInfo, IProgress<string> progress, Ca
                 TaskContinuationOptions.None,
                 TaskScheduler.Current);
 
-            Task<ExportJobOutcome> UpdateExportJob(ExportJobRecord exportJobRecord, WeakETag weakETag, CancellationToken innerCancellationToken)
+            async Task<ExportJobOutcome> UpdateExportJob(ExportJobRecord exportJobRecord, WeakETag weakETag, CancellationToken innerCancellationToken)
             {
                 record = exportJobRecord;
                 progress.Report(JsonConvert.SerializeObject(exportJobRecord));
-                return Task.FromResult(new ExportJobOutcome(exportJobRecord, weakETag));
+
+                if (record.Status != OperationStatus.Completed && record.Status != OperationStatus.Failed)
```

Comment:
Queued shouldn't be possible for a job that is being worked on. When a job is dequeued it is put into the running state.
Running is the expected state for a job that enters this statement.
A job in Canceled state can't be being worked on. That state is only set on jobs in the database.
Paused and Unknown aren't used for Export jobs anywhere.

I could change it to only allow the Running state instead of excluding completed and failed.

Reply:
I like to have only Running

Reply:
Updated